addons/bus/static/src/js/web_client_bus.js

```diff
@@ -0,0 +1,109 @@
+odoo.define('bus.WebClient', function (require) {
+    "use strict";
+
+    const core = require('web.core');
+    const WebClient = require('web.WebClient');
+
+    const _t = core._t;
+
+    WebClient.include({
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * Detects the presence of assets in DOM's HEAD
+         *
+         * @override
+         */
+        async start() {
+            this._assetsChangedNotificationId = null;
+            this._assets = {};
+            this.call('bus_service', 'onNotification', this, this._onNotification);
```

Comment:
you could bind this event handler alongside the registration of "bundle_changed" channel, in show_application.